xcube_sh/constants.py

```diff
@@ -26,6 +26,8 @@
 
 DEFAULT_SH_API_URL = 'https://services.sentinel-hub.com'
 DEFAULT_SH_OAUTH2_URL = f'{DEFAULT_SH_API_URL}/oauth'
+DEFAULT_SH_METADATA_API_URL = f'{DEFAULT_SH_API_URL}/configuration/' \
+                              f'api/v1/metadata/collection/byoc-%s'
```

Comment:
What if the user already includes the "byoc-" prefix in their id? 

Then I get: 
```
---------------------------------------------------------------------------
HTTPError                                 Traceback (most recent call last)
~/Desktop/projects/xcube-sh/xcube_sh/sentinelhub.py in maybe_raise_for_response(cls, response)
    579         try:
--> 580             response.raise_for_status()
    581         except requests.HTTPError as e:

~/miniconda3/envs/xcube-dev/lib/python3.9/site-packages/requests/models.py in raise_for_status(self)
    952         if http_error_msg:
--> 953             raise HTTPError(http_error_msg, response=self)
    954 

HTTPError: 404 Client Error: Not Found for url: https://services.sentinel-hub.com/configuration/api/v1/metadata/collection/byoc-byoc-538fdf6b-d391-4ea8-bba7-47f90026d5e2

The above exception was the direct cause of the following exception:

SentinelHubError                          Traceback (most recent call last)
/tmp/ipykernel_33879/1823333892.py in <module>
----> 1 cube = open_cube(cube_config)
      2 cube

~/Desktop/projects/xcube-sh/xcube_sh/cube.py in open_cube(cube_config, observer, trace_store_calls, max_cache_size, sentinel_hub, **sh_kwargs)
     54     elif sh_kwargs:
     55         raise ValueError(f'unexpected keyword-arguments: {", ".join(sh_kwargs.keys())}')
---> 56     cube_store = SentinelHubChunkStore(sentinel_hub, cube_config, observer=observer,
     57                                        trace_store_calls=trace_store_calls)
     58     if max_cache_size:

~/Desktop/projects/xcube-sh/xcube_sh/chunkstore.py in __init__(self, sentinel_hub, cube_config, observer, trace_store_calls)
    554         self._sentinel_hub = sentinel_hub
    555         if cube_config.band_names is None:
--> 556             bands = sentinel_hub.bands(
    557                 cube_config.dataset_name,
    558                 collection_id=cube_config.collection_id

~/Desktop/projects/xcube-sh/xcube_sh/sentinelhub.py in bands(self, dataset_name, collection_id)
    190             url = DEFAULT_SH_METADATA_API_URL % collection_id
    191             response = self.session.get(url)
--> 192             SentinelHubError.maybe_raise_for_response(response)
    193             return response.json().get('bands', [])
    194 

~/Desktop/projects/xcube-sh/xcube_sh/sentinelhub.py in maybe_raise_for_response(cls, response)
    588             except Exception:
    589                 pass
--> 590             raise SentinelHubError(f'{e}: {detail}' if detail else f'{e}',
    591                                    response=response) from e
    592 

SentinelHubError: 404 Client Error: Not Found for url: https://services.sentinel-hub.com/configuration/api/v1/metadata/collection/byoc-byoc-538fdf6b-d391-4ea8-bba7-47f90026d5e2
```

Reply:
@AliceBalfanz if you have a good reason to use a `byoc-` prefix, then please file a feature request. However, this is not in the scaope of this PR.